apiv6/kentikapi/client_integration_test.go

```diff
@@ -0,0 +1,369 @@
+	"testing"
+	"time"
+
+	"github.com/kentik/community_sdk_golang/apiv6/kentikapi"
+	"github.com/kentik/community_sdk_golang/apiv6/kentikapi/synthetics"
+	"github.com/stretchr/testify/assert"
+)
+
+const (
+	authEmailKey    = "X-CH-Auth-Email"
+	authAPITokenKey = "X-CH-Auth-API-Token"
+	dummyAuthEmail  = "email@example.com"
+	dummyAuthToken  = "api-test-token"
+	testAgentID     = "968"
+)
+
+func TestClient_PatchAgent(t *testing.T) {
+	tests := []struct {
+		name     string
+		retryMax *int
+		request  synthetics.V202101beta1PatchAgentRequest
+		// expectedRequestBody is map for the granularity of assertion diff
+		expectedRequestBody      map[string]interface{}
+		responses                []httpResponse
+		expectedResult           synthetics.V202101beta1PatchAgentResponse
+		expectedError            bool
+		disableHTTPResponseCheck bool
+	}{
+		{
+			name:                "empty request, status Bad Request received",
+			expectedRequestBody: map[string]interface{}{},
+			responses:           []httpResponse{newErrorHTTPResponse(http.StatusBadRequest)},
+			expectedError:       true,
+		}, {
+			name:                "status Bad Request received",
+			request:             newPatchAgentNameRequest(),
+			expectedRequestBody: newPatchAgentNameRequestBody(),
+			responses:           []httpResponse{newErrorHTTPResponse(http.StatusBadRequest)},
+			expectedError:       true,
+		}, {
+			name:                "name updated",
+			request:             newPatchAgentNameRequest(),
+			expectedRequestBody: newPatchAgentNameRequestBody(),
+			responses: []httpResponse{{
+				statusCode: http.StatusOK,
+				body:       dummyAgentResponseBody,
+			}},
+			expectedResult: synthetics.V202101beta1PatchAgentResponse{Agent: newDummyAgent()},
+		}, {
+			name:                "retry till success when status Too Many Requests received",
+			request:             newPatchAgentNameRequest(),
+			expectedRequestBody: newPatchAgentNameRequestBody(),
+			responses: []httpResponse{
+				newErrorHTTPResponse(http.StatusTooManyRequests),
+				newErrorHTTPResponse(http.StatusTooManyRequests),
+				newErrorHTTPResponse(http.StatusTooManyRequests),
+				{
+					statusCode: http.StatusOK,
+					body:       dummyAgentResponseBody,
+				},
+			},
+			expectedResult: synthetics.V202101beta1PatchAgentResponse{Agent: newDummyAgent()},
+		}, {
+			name:                "retry 4 times when status 429 or 5xx received (except 501)",
+			request:             newPatchAgentNameRequest(),
+			expectedRequestBody: newPatchAgentNameRequestBody(),
+			responses: []httpResponse{
+				newErrorHTTPResponse(http.StatusInternalServerError),
+				newErrorHTTPResponse(http.StatusBadGateway),
+				newErrorHTTPResponse(http.StatusServiceUnavailable),
+				newErrorHTTPResponse(http.StatusGatewayTimeout),
+				newErrorHTTPResponse(http.StatusTooManyRequests),
+			},
+			expectedError: true,
+		}, {
+			name:                "retry 4 times when status 429 or 5xx received (except 501) and last status is 5xx",
```

Comment:
I don't quite understand the "5xx" part of the comment.

Reply:
It means codes from 500 to 599. I'll change the default behaviour in next PR, so this name will change